zubhub_frontend/zubhub/src/components/hamburger_menu/HamburgerSidebar.jsx

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import styles from '../../assets/js/styles/components/hamburger_menu/hamburgerSidebarStyles.js';
+import commonStyles from '../../assets/js/styles';
+import { makeStyles } from '@material-ui/core/styles';
+import {
+  logout,
+  handleHamburgerMenuClose,
+} from '../../views/pageWrapperScripts';
+import clsx from 'clsx';
+import cn from 'classnames';
+
+import {
+  Typography,
+  Drawer,
+  Link,
+  Avatar,
+  MenuItem,
+} from '@material-ui/core';
+
+const useStyles = makeStyles(styles);
+const useCommonStyles = makeStyles(commonStyles);
+
+const HamburgerSidebar = ({
+  hamburgerMenuOpen,
+  setState,
+  t,
+  props,
```

Comment:
In order to prevent prop drilling (https://www.geeksforgeeks.org/what-is-prop-drilling-and-how-to-avoid-it/) we want to use hooks when possible. Redux has hooks to get access to global state and dispatch functions (https://react-redux.js.org/api/hooks). This should allow you to get access to ```auth``` information. In addition for the ```t``` function ```i18next``` has hooks (https://react.i18next.com/latest/usetranslation-hook) for accessing that as well so we shouldn't need to pass it in through props.